webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -620,60 +620,100 @@ private static void addScrambles(PdfWriter docWriter, Document doc, ScrambleRequ
-                font.setSize(size);
-                ct.setSimpleColumn(new Phrase(text, font), rect.getLeft(), rect.getBottom(), rect.getRight(), rect.getTop(), size, Element.ALIGN_LEFT);
-                ct.setRunDirection(runDirection);
-                status = ct.go(true);
-                if ((status & ColumnText.NO_MORE_TEXT) != 0 && (newlinesAllowed || ct.getLinesWritten() <= 1)) {
-                    if (max - min < size * precision) {
-                        return size;
-                    }
-                    min = size;
+
+            //first make the chunks using a potentialFontSize
+            //starting with the maximum size
+            float oldFont = minFont;
+            float newFont = maxFont;
+
+            //determine line height - using font ascent and descent
+            float lineHeight = font.getBaseFont().getAscentPoint(text, newFont) - font.getBaseFont().getDescentPoint(text, newFont);
+            
+            font.setSize(newFont);
+
+            LinkedList<Chunk> lineChunks = splitScrambleToLineChunks(text, font, rect.getWidth());
+
+            int nLines = lineChunks.size();
+
+            //line spacing is 0.5*font by default
+            float totalHeight = (float)nLines * lineHeight * 1.5f - (0.5f * lineHeight);
+
+            boolean fits = (totalHeight < rect.getHeight());
+
+            float diff = Math.abs(newFont - oldFont);
+
+            boolean stop = false;
+
+            if(diff < precision && fits) {
+                stop = true;
+            }
+
+            //loop until it's good enough
+            while(!stop) {
+
+                if(totalHeight < rect.getHeight()) {
+                    minFont = newFont;
                 } else {
-                    max = size;
+                    maxFont = newFont;
+                }
+
+                oldFont = newFont;
+
+                newFont = (float)(maxFont + minFont) / 2.0f;
+
+                lineHeight = font.getBaseFont().getAscentPoint(text, newFont) - font.getBaseFont().getDescentPoint(text, newFont);
+
+                font.setSize(newFont);
+
+                lineChunks = splitScrambleToLineChunks(text, font, rect.getWidth());
+
+                nLines = lineChunks.size();
+
+                totalHeight = (float)nLines * lineHeight * 1.5f - (0.5f * lineHeight);
+
+                fits = (totalHeight < rect.getHeight());
+
+                diff = Math.abs(newFont - oldFont);
+
+                if(diff < precision && fits) {
+                    stop = true;
                 }
             }
-            return size;
-        } catch (Exception e) {
-            throw new com.itextpdf.text.ExceptionConverter(e);
+
+            return newFont;
+           
+        } else { //trying to fit in a single line
+            //Again, char width is fontSize * 0.6, but it may get too big (skewb specially)
+            //so we use 0.65 instead
+            float fontSize = (rect.getWidth() / (float)text.length()) * 0.65f;
```

Comment:
What's going on here? The comment doesn't make it clear to me why you need 0.65 here instead of 0.6.